locator/network_topology_strategy.cc

```diff
@@ -554,6 +556,35 @@ tablet_replica_set network_topology_strategy::drop_tablets_in_dc(schema_ptr s, c
     return filtered;
 }
 
+void network_topology_strategy::sanity_check_read_replicas(const effective_replication_map& erm,
+                                                           const inet_address_vector_replica_set& read_replicas) const {
+    const auto& topology = erm.get_topology();
+
+    struct rf_node_count {
+        size_t replication_factor{0};
+        size_t node_count{0};
+    };
+
+    boost::unordered_flat_map<sstring, rf_node_count> data_centers_replication_factor;
```

Comment:
What is the advantage of using it here over `std::unordered_map`? The disadvantage is that that will pull yet more bunch of code as a dependency.

Reply:
But we already have `boost` as dependency. i'd say using unordered_flat_map (or another open addressed hashmap) should be default in high performance code. the only possible issue is that the iterators will be invalidated in case of rehashing.

Reply:
Boost has a lot of things. We do not use all of them. We should always prefer std over it if possible. This is not a high performance code since we do not run it in release, no? May be we should run it in debug only in fact. I do not think such relatively slow checks should be part of dev. And in case perf is needed why not use `absl::flat_hash_map` which is already used by the project? But again if perf is not needed lets just use std.

Reply:
I'm not sure I completely understand your concerns. Is it a possible issue with stability, since this `boost`  container is not used in the project?  I can change it to `absl::` version if it helps. I mean even if this is debug code, why pessimize its performance?

Reply:
No, not stability just more dependency bloat without real reason. Also developers familiar with std and it is well documented (unlike boost). There are hundreds (I am not joking) variants of containers for c++. If each developer will use his/her favorite the code will be a mess. I do not mind `boost::unordered_flat_map` in performance critical code if it is better then `absl::` but then why not replace existing `absl::` with it? But here I do not even see why bother with non std. If you still insist it is a performance thing please use `absl::`. And if there is a independent benchmark that shows that  `boost::unordered_flat_map` is strictly better lets replace `absl::` with it as a follow up everywhere.

Reply:
It's just my habit to use `boost` and I see `boost` is already a dependency, so I did not think it's a big issue if I use a container from a library already used in the project. I added it specifically in a .cc module to reduce the scope and the impact of the compilation time. 
I'm not insisting that `absl::flat_hash_map` will be faster than `boost::unordered_flat_map`, I'd expect these would be similar in performance as these are built basing on the same idea (and the tests [show ](https://bannalia.blogspot.com/2022/11/inside-boostunorderedflatmap.html#:~:text=As%20predicted%20by%20our%20statistical,first%20check%20that%20the%20element) a slight advantage of `boost::unordered_falt_map`). But I'm pretty sure they will outperform `std::unordered_map` and any other closed addressed hash map.